### DISCUSSION TITLE
docker image

### DISCUSSION THREAD

**Discussion by MyraBaba** [obss/sahi#614] [Ideas]
Hi,

do you have docker image for easy test and fast cold start ?

**Comment by fcakyon** [comment]
Hello @MyraBaba we currently don't have a ready dockerfile but it is easy to create one. Feel free to open a PR and we will accept it.

You can create a docker file that starts from a `11.3.0-cudnn8-runtime-ubuntu20.04` base image and add these extra installations:

- Install `sahi` using pip:

```console
pip install sahi
```

- Install your desired version of pytorch and torchvision:

```console
pip install pytorch=1.10.2 torchvision=0.11.3
```
  
- Install your desired detection framework (yolov5):

```console
pip install yolov5==6.1.3
```

- Install your desired detection framework (mmdet):

```console
pip install mmcv-full==1.5.3 -f https://download.openmmlab.com/mmcv/dist/cu113/torch1.10.0/index.html
```

```console
pip install mmdet==2.25.0
```

- Install your desired detection framework (detectron2):

```console
pip install detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/cu113/torch1.10/index.html
```

- Install your desired detection framework (huggingface):

```console
pip install transformers timm
```

**Comment by MyraBaba** [comment]
@fcakyon  to reduce to friction for new comer it will be very useful a fully working docker image. 
ie. :

`sahi predict --model_path /home/alp2080/Projects/yolov5/yolov5s.pt --model_type yolov5 --source /home/xxx/DATA/Videos/hiv00038.mp4`

now gives segmentation fault  error(core dumped)

**Comment by devrimcavusoglu** [comment]
> @fcakyon to reduce to friction for new comer it will be very useful a fully working docker image. ie. :
> 
> `sahi predict --model_path /home/alp2080/Projects/yolov5/yolov5s.pt --model_type yolov5 --source /home/xxx/DATA/Videos/hiv00038.mp4`
> 
> now gives segmentation fault error(core dumped)

Hi @MyraBaba, can you please provide your platform information (OS, python, sahi versions, etc.) ? and also is that possible you can create a minimal reproducible example with some small video (so that we can try to recreate the problem you encounter) ?

**Comment by MyraBaba** [comment]
Ubuntu 20.04 ,  2080rtx ti , latest sahi , python3.8 will try findout.

Additionally; when its working ı noticed that sahi is very slow , but gPU is utilzed %20.. FYI
